tests/ga/test_extension.py

```diff
@@ -79,103 +81,129 @@ def raise_ioerror(*args):
-
-            dst = os.path.join(self.lib_dir,
-                               handler.get_full_name() + HANDLER_PKG_EXT)
-            shutil.copy(src, dst)
-
-            if not handler_state is None:
-                zipfile.ZipFile(dst).extractall(handler.get_base_dir())
-                handler.set_handler_state(handler_state)
+        self.mock_sleep = patch("time.sleep", lambda *_: mock_sleep(0.0001))
+        self.mock_sleep.start()
 
-            version += 1
+    def tearDown(self):
+        AgentTestCase.tearDown(self)
+        self.mock_sleep.stop()
 
-    def _count_packages(self):
-        return len(glob.glob(os.path.join(self.lib_dir, "*.zip")))
-
-    def _count_installed(self):
-        paths = os.listdir(self.lib_dir)
-        paths = [os.path.join(self.lib_dir, p) for p in paths]
-        return len([p for p in paths
-                    if os.path.isdir(p) and self._is_installed(p)])
+    @staticmethod
+    def _count_packages():
+        return len(glob.glob(os.path.join(conf.get_lib_dir(), "*.zip")))
 
-    def _count_uninstalled(self):
-        paths = os.listdir(self.lib_dir)
-        paths = [os.path.join(self.lib_dir, p) for p in paths]
+    @staticmethod
+    def _count_extension_directories():
+        paths = [os.path.join(conf.get_lib_dir(), p) for p in os.listdir(conf.get_lib_dir())]
         return len([p for p in paths
-                    if os.path.isdir(p) and not self._is_installed(p)])
+                    if os.path.isdir(p) and TestExtensionCleanup._is_extension_dir(p)])
 
-    def _is_installed(self, path):
-        path = os.path.join(path, 'config', 'HandlerState')
-        return fileutil.read_file(path) != "NotInstalled"
-
-    @patch("azurelinuxagent.common.conf.get_lib_dir")
-    def test_cleanup_leaves_installed_extensions(self, mock_conf):
-        mock_conf.return_value = self.lib_dir
+    @staticmethod
+    def _is_extension_dir(path):
+        return re.match(HANDLER_NAME_PATTERN, os.path.basename(path))
 
-        self._install_handlers(start=0, count=5, handler_state=ExtHandlerState.Installed)
-        self._install_handlers(start=5, count=5, handler_state=ExtHandlerState.Enabled)
 
-        self.assertEqual(self._count_packages(), 10)
-        self.assertEqual(self._count_installed(), 10)
+    def _assert_ext_handler_status(self, report_vm_status, expected_status, version, expected_ext_hanlder_count=0):
+        self.assertTrue(report_vm_status.called)
+        args, kw = report_vm_status.call_args
+        vm_status = args[0]
+        self.assertEqual(expected_ext_hanlder_count, len(vm_status.vmAgent.extensionHandlers))
+        for ext_handler in vm_status.vmAgent.extensionHandlers:
 
-        self.ext_handlers._cleanup_outdated_handlers()
+            self.assertEquals(expected_status, ext_handler.status)
+            self.assertEquals(version, ext_handler.version)
+            # self.assertEquals(expected_handler_name, handler_status.name)
```

Comment:
Remove?

Reply:
Rather, uncomment?

Reply:
Removed it, we dont really need it. Mainly just checking if the cleanup was successful or not.